numpy/random/tests/test_random.py

```diff
@@ -255,6 +255,18 @@ def test_random_integers_max_int(self):
         desired = np.iinfo('l').max
         np.testing.assert_equal(actual, desired)
 
+    def test_random_integers_deprecated(self):
```

Comment:
I'm -0 on adding `_DeprecationTestCase` to the *public* API, because it feels a bit underbaked (`assert_deprecated`'s calling convention is super complicated!), so probably we shouldn't set it in stone as-is.

No objections to refactoring our test suite to have more useful internal tools though. Or maybe `assert_warns` should be clever enough to set up the warning context properly, if it isn't already.